clip.cpp

```diff
@@ -15,13 +15,20 @@
 
 // utility function for a workaround until https://github.com/ggerganov/ggml/issues/260 is resolved
 // after that, remove this and use the mechanism implemented in GGML directly
-size_t get_mem_req_by_size(size_t n_tensors)
+size_t get_mem_req_by_size(const size_t n_tensors, const int n_image_positions)
 {
     size_t mb = 1024 * 1024;
     switch (n_tensors)
     {
     case 397: // base
-        return 8 * mb;
+        if (n_image_positions == 50)
```

Comment:
why use `==` here and `<=` down below.
also why different param name?

Reply:
We call this function only once in `clip_model_load` and allocate that memory based the number of positions in the vision model. The scratch buffer, however, is allocated separately in `clip_image_encode` and `clip_text_encode`. `n_image_positions` is fixed for a gien model, but `n_positions` might be different in the case of the text model. So if I used `==` there, the condition would not return `true` for shorter texts.